src/Interpreters/Cache/Metadata.cpp

```diff
@@ -515,7 +513,7 @@ friend struct CacheMetadata;
     std::queue<DownloadInfo> queue;
 };
 
-void CacheMetadata::downloadThreadFunc()
+void CacheMetadata::downloadThreadFunc(const bool & stop_flag)
```

Comment:
You mean better by copy? In this case copy won't work because this bool flag can be set in another thread and we want to be able to see that change. (This bool is not atomic, because it is updated and checked under mutex anyway).